src/sentry/api/endpoints/organization_release_details.py

```diff
@@ -527,14 +527,21 @@ def put(self, request, organization, version):
 
             refs = result.get("refs")
             if not refs:
-                refs = [
-                    {
-                        "repository": r["repository"],
-                        "previousCommit": r.get("previousId"),
-                        "commit": r["currentId"],
-                    }
-                    for r in result.get("headCommits", [])
-                ]
+                # Handle legacy
+                if result.get("headCommits", []):
```

Comment:
nit: You don't need the default value param (`[]`) since this is checking if there is a value in the first place. If there isn't a `headCommits` key, it would return `None` which fails the conditional the same way `[]` would.

Reply:
@leeandher I think I'll keep the default bc it defines in code that the legacy `headCommits` should be an array.